test_conformance/math_brute_force/unary.cpp

```diff
@@ -506,14 +506,15 @@ static cl_int TestFloat( cl_uint job_id, cl_uint thread_id, void *data )
         float p_j = *(float *) &p[j];
         if ( strcmp(fname,"sin")==0 || strcmp(fname,"cos")==0 )  //the domain of the function is [-pi,pi]
         {
-          if( fabs(p_j) > M_PI )
-            p[j] = NAN;
+            if (fabs(p_j) > M_PI) ((float *)p)[j] = NAN;
         }
 
         if ( strcmp( fname, "reciprocal" ) == 0 )
         {
-          if( fabs(p_j) > 0x7E800000 ) //the domain of the function is [2^-126,2^126]
-            p[j] = NAN;
+            cl_uint limit = 0x7E800000;
+            if (fabs(p_j) > *((float *)&limit)) // the domain of the function is
+                                                // [2^-126,2^126]
```

Comment:
We're missing equivalent code to exclude small numbers from the domain

Reply:
Fixed.